### PR TITLE
Fix idna bug in requires_non_ascii

### LINKED ISSUE #155: Unexpected exception from EmailAddress.full_spec
Found in version: 0.6.8

It is possible for `EmailAddress.full_spec` function to raise an exception even if `EmailAddress.require_non_ascii` returns `False`, see a stack trace below:
```
Traceback (most recent call last):
  File "/app/src/flanker/flanker/mime/message/headers/encoding.py", line 37, in encode
    return encode_unstructured(name, value)
  File "/app/src/flanker/flanker/mime/message/headers/encoding.py", line 52, in encode_unstructured
    return encode_address_header(name, value)
  File "/app/src/flanker/flanker/mime/message/headers/encoding.py", line 65, in encode_address_header
    out.append(addr.full_spec().encode('utf-8'))
  File "/app/src/flanker/flanker/addresslib/address.py", line 536, in full_spec
    ace_hostname = self.hostname.encode('idna')
  File "/usr/lib/python2.7/encodings/idna.py", line 164, in encode
    result.append(ToASCII(label))
  File "/usr/lib/python2.7/encodings/idna.py", line 76, in ToASCII
    label = nameprep(label)
  File "/usr/lib/python2.7/encodings/idna.py", line 38, in nameprep
    raise UnicodeError("Invalid character %r" % c)
UnicodeError: Invalid character u'\u200e'
```

That breaks the mutual contract of these functions.

### PR DESCRIPTION
**Pull request by b0d0nne11**
Resolves #155

### CODE REVIEW

**Comment by b0d0nne11** [comment]
@horkhe do you have an example of a hostname that fails IDNA encoding? The one example I see in the IDNA docs passes because we lowercase domains and that resolves the encoding issue.

**Comment by horkhe** [comment]
@b0d0nne11 sorry, but I do not have an example

File: flanker/addresslib/address.py [hunk 1]
```diff
@@ -535,7 +535,11 @@ def full_spec(self):
         if not is_pure_ascii(self.mailbox):
```
Comment (horkhe) [inline flanker/addresslib/address.py]: I think `is_pure_ascii` should return `False` if the domain part cannot be IDNA encoded. There should be a contract that if `is_pure_ascii` returns `True` it should be safe to call `EmailAddress.full_spec`.

Comment (horkhe) [inline flanker/addresslib/address.py]: Scratch that I mixed up `is_pure_ascii` and `EmailAddress.requires_non_ascii`.

**Pull request merged**